micrometer-commons/src/main/java/io/micrometer/common/lang/internal/Contract.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2025 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.common.lang.internal;
+
+import java.lang.annotation.*;
+
+/**
+ * This is for internal use only.
```

Comment:
Maybe worth writing more here, but I wanted to be very clear this is not supported and should not be used outside of Micrometer. My hope is that we can remove it in the future, and so I want to be clear this is not subject to normal compatibility guarantees. Maybe we can even mark is deprecated if that doesn't make it annoying for us to use.

Reply:
I think this is fine. I'm thinking if we should add a link in a non-javadoc comment to nullaways' doc about this in case someone is working on Micrometer's codebase and wanders here.